src/item_location.cpp

```diff
@@ -161,7 +166,11 @@ class item_location::item_on_person : public item_location::impl
                 // item already in target characters inventory at base of stack
                 return ch.get_item_position( it );
             } else {
-                return ch.get_item_position( &ch.i_add( who.i_rem( it ) ) );
+                item obj = what->split( qty );
+                if( !obj.is_null() ) {
```

Comment:
This looks ... wrong. The item is removed if splitting the stack succeeded, which implies there is still something of the original stack left.

In `item_location::item_on_map`, it's the other way: if splitting return non-null, return early. If splitting returns null, call `remove_item`.